### DISCUSSION TITLE
Apple Pencil should be registered as touch input.

### DISCUSSION THREAD

**Discussion by Axcxiii** [ppy/osu#27437] [Ideas]
The title explains it. It would be nice because the pencil being the cursor is kinda weird in my case.

**Comment by cihe13375** [comment]
I have no apple devices but I don't think it's a good idea, if it means everyone playing with apple pencil would be applied the TD mod (even though the cursor can't really "teleport").

**Reply by Rekunan** [reply]
There is still a massive advantage in that you no longer have the mental block between mapping the movement of your input device to your cursor movement.

**Reply by cihe13375** [reply]
yea, but it also has significant disadvantages (think about making the tablet 10x larger and play with full area). Also it is already possible in stable anyway (with a pen tablet or such), and there are good reasons why it does not become the mainstream. Personally I think this play style is fun and intuitive to new players, but is not really beneficial in a competitive context.

**Reply by Rekunan** [reply]
I think the disadvantages actually aren't that significant. Think about how almost every digital artist prefers a screen tablet over a pen tablet. I can also bring up my own anecdotal experience from the arcade, where I always perform better on cabinets with the mental mapping requirement removed, such as chunithm, maimai, and jubeat, compared to ones without it removed, such as popn, sdvx, and iidx.

People also consider Taiko to be extremely reading heavy, and that's because the only indication for the notes is color, nothing about position, and is why taikomania and all of its implications and stigma exist.

You are correct that there are good reasons why it isn't mainstream, and that's because of the overall cost (not just in money) of touchscreen monitors. However, this could very soon change as the mobile market grows. This also brings up a good point about the ease of mobile rhythm games compared to non-mobile, especially direct counterparts/ports such as FNF.

**Reply by eclecticfluff** [reply]
Digital artists don′t move their stylus at speed or with abrupt changes in direction. Can you point to a Chunithm, Maimai, or Jubeat chart that requires moving one hand quickly between different points?I at least know one such case fron Arcaea in [Pentiment](https://m.youtube.com/watch?v=DzZ1IN3BLI0&t=2m24s) and let me tell you, off owie my stamina. 
Some other thoughts too: It′s not like the game hasn′t accepted other improvements in technology before that help players be better before. Besides, considering this game′s origins as an EBA simulator, wouldn′t using a pen tablet with a screen get closest to the origonal experience it was trying to replicate? Limiting how competitive a huge new wave of players can be, many of whome likely will have no way to play on computer feels like a harsh stance. Again, PC players could be using screen tablets like artists but they don′t. It can′t be a problem of money considering how spendy people can already be on top end gear. So, why is that?

**Reply by Rekunan** [reply]
> that requires moving one hand quickly between different points

Not all osu maps require this, especially now with most of the recent ranked section requiring fluid motions, and I won't provide because I do agree that those kinds of charts require a lot more stamina, however, I think to look at the true advantage you should look at [bonk](https://osu.ppy.sh/users/4209965), who has been using this play style. He has been known in the community for having insane aim control skill, and yet, isn't it interesting to see how low his hours and playcount are? Especially considering how aim control requires a lot more reading, or at least a different kind, compared to standard aim.

> Limiting how competitive ... feels like a harsh stance

I agree that it's a harsh stance, but I think it's necessary, esp considering it's one of the few rhythm games to have a pc and mobile port. It's also not really limiting how competitive mobile players can be, it's just putting them on the same playing field as pc players.

> considering how spendy people can already be on top end gear

This is underestimating the sheer amount of laptop players who can't even afford a pc. However, for the people who can, you are right that money wouldn't be an issue, and most likely the same with other issues such as setup, reliability, etc. The main thing would be the mental block, as most people have the same misconception as you, thinking that there's no advantage, so they don't bother to try it out. And I don't mean to downplay the mental block, as there are good reasons for it existing, such as it being the popular community opinion.

**Reply by KevinL88888** [reply]
I play with an iPad pro using a stylus to drag the cursor and I adjusted the gameplay area to be 130 mm by 130 mm(basically tablet full area) and is super fun to play osu this way. I just hope they change the pp nerf for touchscreen players that use a drag play style because playing by dragging the cursor is just as playing with mouse or drawing tablet.

**Reply by KevinL88888** [reply]
And I also have a gamer laptop with a drawing tablet where I have tried osulazer but I always comeback to iPad because is just more fun. I feel I'm more immerse in the gameplay that way

**Comment by KevinL88888** [comment]
Not a good idea. I literally just bought an Apple pencil so I can play by dragging without having my pp nerf. Playing in iPad pro is so much fun and I don't want dragging play style to be nerf.

**Comment by KevinL88888** [comment]
The Apple pencil is already registered as a touch input... :(
And I literally bought it today thinking that way my pp wouldn't be nerf by using dragging play style. 

Where is Bonk when we need him to defend his play style!?

**Comment by P4blohh** [comment]
Just thought i would chime in on this, since it seems that a large portion of people are unaware of just how much TDP (Touch Device Pen) differs from more conventional playstyles such as mouse and tablet. Here is a short text i wrote regarding a player named Bonk, who utilizes TDP (although he uses his finger instead of a pen):
"Mechanically it might seem that his playstyle functions the same way as other conventional playstyles (mouse and tablet), but fundamentally his playstyle bypasses an entire portion of the reading process that is unique to those playstyles, and for that reason alone it makes no sense for him to be on the same leaderboard as mouse and tablet players. It cannot be understated how crucial the degree of separation between your cursor and input device is, as it requires you to translate your reading into tablet / mouse movements that THEN get translated into cursor movements. This is both a mental and physical process that requires a HUGE deal of muscle memory to perfect. Bonks playstyle skips that process almost entirely. His input device allows him to aim the notes directly, as his hand movements don’t have to go through a medium before getting translated into cursor movements. In other words, his input device movements ARE his cursor movements. Not only does this undermine the perceived difficulty of many of the aim control maps he's playing, but it's also just an entirely different playstyle that differs heavily from mouse and tablet. In conclusion, they aren't legit if you're comparing them to plays set on mouse and tablet"
Now obviously, just like with any other playstyle, TDP does have it's own downsides, and while some of those downsides are pretty challenging to deal with, i don't believe they come anywhere near close to offsetting the one monumental change that comes with TDP. I personally believe players using TDP should have their own separate leaderboards, as i find it nonsensical to have fundamentally different playstyles on the same leaderboard.

**Comment by Avelia-tr** [comment]
Bad idea imo it would mean tablet with screen needs to also be detected as such wich seems not doable

**Comment by Espionage724** [comment]
I don't think it should be touch input; this isn't much different than a PC set-up (iPad, Pencil, BT KB): https://www.youtube.com/watch?v=afE5ekOd1Xg

_This_ is touch 😎 https://www.youtube.com/watch?v=mAO_ObZGDXo
